bundles/AdminBundle/Resources/public/js/pimcore/asset/listfolder.js

```diff
@@ -154,35 +269,153 @@ pimcore.asset.listfolder = Class.create({
+                       renderer: function(d) {
+                            var date = new Date(d * 1000);
+                            return Ext.Date.format(date, "Y-m-d H:i:s");
                         }
-                    }.bind(this)
+                    });
+                } else if (field.key == "filename") {
+                    gridColumns.push({text: t(field.key), sortable: true, dataIndex: field.key, editable: false,
+                        width: this.getColumnWidth(field, 250), filter: 'string', renderer: Ext.util.Format.htmlEncode});
+                } else if (field.key == "fullpath") {
+                    gridColumns.push({text: t(field.key), sortable: true, dataIndex: field.key, editable: false,
+                        width: this.getColumnWidth(field, 400), filter: 'string', renderer: Ext.util.Format.htmlEncode});
+                } else if (field.key == "size") {
+                    gridColumns.push({text: t(field.key), sortable: false, dataIndex: field.key, editable: false,
+                        width: this.getColumnWidth(field, 130)});
+                } else {
+                    gridColumns.push({text: t(field.key),  width: this.getColumnWidth(field, 130), sortable: true,
+                        dataIndex: field.key});
                 }
-            ]
-        });
+            } else if (field.type == "date") {
+                gridColumns.push({text: field.key,  width: this.getColumnWidth(field, 120), sortable: false,
+                    dataIndex: field.key, filter: 'date', editable: false,
+                    renderer: function(d) {
+                        if (d) {
+                            var date = new Date(d * 1000);
+                            return Ext.Date.format(date, "Y-m-d");
+                        }
 
-        this.grid.on("rowcontextmenu", this.onRowContextmenu);
+                    }
+                });
+            } else if (field.type == "checkbox") {
+                gridColumns.push(new Ext.grid.column.Check({
+                    text:  field.key,
+                    editable: false,
+                    width: this.getColumnWidth(field, 40),
+                    sortable: false,
+                    filter: 'boolean',
+                    dataIndex: field.key
+                }));
+            } else if (field.type == "select") {
+                gridColumns.push({text: field.key,  width: this.getColumnWidth(field, 200), sortable: false,
+                    dataIndex: field.key, filter: 'string'});
+            } else if (field.type == "document" || field.type == "asset" || field.type == "object") {
+                gridColumns.push({text: field.key,  width: this.getColumnWidth(field, 300), sortable: false,
+                    dataIndex: field.key});
+            } else {
+                gridColumns.push({text: field.key,  width: this.getColumnWidth(field, 250), sortable: false,
+                    dataIndex: field.key, filter: 'string'});
```

Comment:
just a thought: for localized metadata: can we show the language flag + key instead of concatenated key~~language ?